Can't change Setting on VM with Custom Unconstrained Compute Offering

##### ISSUE TYPE
 * Bug Report

##### CLOUDSTACK VERSION
~~~
4.16.0
~~~

##### CONFIGURATION
XCP-NG 8.2
Advanced Zone

##### OS / ENVIRONMENT
Oracle Linux 8.4 - Management server, MYSQL 8.0.26
XCP-NG 8.2 - hypervisor

##### SUMMARY
Can't change Setting on VM with Custom Unconstrained Compute Offering

##### STEPS TO REPRODUCE
~~~
Deploy VM with Custom Unconstrained Compute Offering. Choose correct CPU Cores, CPU Speed in MHz (for example 2000), RAM capasity. Poweroff VM from WebUI. Try to change VM's settings. 
~~~
##### EXPECTED RESULTS
~~~
User or Admin can change VM's settings.
~~~

##### ACTUAL RESULTS
~~~
When U try to change any setting U have error about CPU Speed. User don't have any errors with Custom Constrained Compute Offering
~~~
![image](https://user-images.githubusercontent.com/42668237/142235062-8b3b1237-9bb1-4c83-be64-567acff37ea2.png)

ERROR log:


2021-11-17 18:57:56,757 ERROR [c.c.a.ApiServer] (qtp239372207-24325:ctx-db2e8cf1 ctx-ecd4e31b) (logid:69235cf3) unhandled exception executing api command: [Ljava.lang.String;@2fa01244
com.cloud.utils.exception.CloudRuntimeException: Failed to update VM, due to: Invalid cpu speed value, specify a value between 1 and 2147483647
        at org.apache.cloudstack.api.command.user.vm.UpdateVMCmd.execute(UpdateVMCmd.java:271)
        at com.cloud.api.ApiDispatcher.dispatch(ApiDispatcher.java:156)
        at com.cloud.api.ApiServer.queueCommand(ApiServer.java:772)
        at com.cloud.api.ApiServer.handleRequest(ApiServer.java:596)
        at com.cloud.api.ApiServlet.processRequestInContext(ApiServlet.java:321)
        at com.cloud.api.ApiServlet$1.run(ApiServlet.java:134)
        at org.apache.cloudstack.managed.context.impl.DefaultManagedContext$1.call(DefaultManagedContext.java:55)
        at org.apache.cloudstack.managed.context.impl.DefaultManagedContext.callWithContext(DefaultManagedContext.java:102)
        at org.apache.cloudstack.managed.context.impl.DefaultManagedContext.runWithContext(DefaultManagedContext.java:52)
        at com.cloud.api.ApiServlet.processRequest(ApiServlet.java:131)
        at com.cloud.api.ApiServlet.doGet(ApiServlet.java:93)
        at javax.servlet.http.HttpServlet.service(HttpServlet.java:645)
        at javax.servlet.http.HttpServlet.service(HttpServlet.java:750)
        at org.eclipse.jetty.servlet.ServletHolder$NotAsync.service(ServletHolder.java:1443)
        at org.eclipse.jetty.servlet.ServletHolder.handle(ServletHolder.java:791)
        at org.eclipse.jetty.servlet.ServletHandler.doHandle(ServletHandler.java:550)
        at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:143)
        at org.eclipse.jetty.security.SecurityHandler.handle(SecurityHandler.java:602)
        at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:127)
        at org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:235)
        at org.eclipse.jetty.server.session.SessionHandler.doHandle(SessionHandler.java:1624)
        at org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:233)
        at org.eclipse.jetty.server.handler.ContextHandler.doHandle(ContextHandler.java:1435)
        at org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:188)
        at org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:501)
        at org.eclipse.jetty.server.session.SessionHandler.doScope(SessionHandler.java:1594)
        at org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:186)
        at org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:1350)
        at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:141)
        at org.eclipse.jetty.server.handler.gzip.GzipHandler.handle(GzipHandler.java:766)
        at org.eclipse.jetty.server.handler.HandlerCollection.handle(HandlerCollection.java:146)
        at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:127)
        at org.eclipse.jetty.server.Server.handle(Server.java:516)
        at org.eclipse.jetty.server.HttpChannel.lambda$handle$1(HttpChannel.java:388)
        at org.eclipse.jetty.server.HttpChannel.dispatch(HttpChannel.java:633)
        at org.eclipse.jetty.server.HttpChannel.handle(HttpChannel.java:380)
        at org.eclipse.jetty.server.HttpConnection.onFillable(HttpConnection.java:273)
        at org.eclipse.jetty.io.AbstractConnection$ReadCallback.succeeded(AbstractConnection.java:311)
        at org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:105)
        at org.eclipse.jetty.io.ChannelEndPoint$1.run(ChannelEndPoint.java:104)
        at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.runTask(EatWhatYouKill.java:336)
        at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.doProduce(EatWhatYouKill.java:313)
        at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.tryProduce(EatWhatYouKill.java:171)
        at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.run(EatWhatYouKill.java:129)
        at org.eclipse.jetty.util.thread.ReservedThreadExecutor$ReservedThread.run(ReservedThreadExecutor.java:375)
        at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:773)
        at org.eclipse.jetty.util.thread.QueuedThreadPool$Runner.run(QueuedThreadPool.java:905)
        at java.base/java.lang.Thread.run(Thread.java:829)
Caused by: com.cloud.exception.InvalidParameterValueException: Invalid cpu speed value, specify a value between 1 and 2147483647
        at com.cloud.vm.UserVmManagerImpl.validateCustomParameters(UserVmManagerImpl.java:1205)
        at com.cloud.vm.UserVmManagerImpl.verifyVmLimits(UserVmManagerImpl.java:2630)
        at com.cloud.vm.UserVmManagerImpl.updateVirtualMachine(UserVmManagerImpl.java:2736)
        ... 62 more
2021-11-17 18:57:56,761 DEBUG [c.c.a.ApiServlet] (qtp239372207-24325:ctx-db2e8cf1 ctx-ecd4e31b) (logid:69235cf3) ===END===  192.168.62.167 -- GET  id=eb4fc14e-7ded-4db5-97da-e24235d324ed&details[0].cpuNumber=1&details[0].memory=1024&details[0].cpuSpeed=2000&details[0].memoryOvercommitRatio=1.0&details[0].Message.ReservedCapacityFreed.Flag=false&details[0].hypervisortoolsversion=xenserver61&details[0].cpuOvercommitRatio=2.0&details[0].platform=device-model:qemu-upstream-compat%3Bvga:std%3Bvideoram:8%3Bapic:true%3Bdevice_id:0001%3Bviridian:false%3Btimeoffset:0%3Bpae:true%3Bacpi:1%3Bhpet:true%3Bsecureboot:false%3Bnx:true&command=updateVirtualMachine&response=json

--- a/server/src/main/java/com/cloud/vm/UserVmManagerImpl.java
+++ b/server/src/main/java/com/cloud/vm/UserVmManagerImpl.java
@@ -2644,6 +2644,7 @@ public class UserVmManagerImpl extends ManagerBase implements UserVmManager, Vir
             Map<String, String> customParameters = new HashMap<>();
             customParameters.put(VmDetailConstants.CPU_NUMBER, String.valueOf(newCpu));
             customParameters.put(VmDetailConstants.MEMORY, String.valueOf(newMemory));
+            customParameters.put(VmDetailConstants.CPU_SPEED, details.get(VmDetailConstants.CPU_SPEED));
             validateCustomParameters(svcOffering, customParameters);
         }
         if (VirtualMachineManager.ResourceCountRunningVMsonly.value()) {

